pkg/common/client.go

```diff
@@ -19,172 +18,117 @@ import (
 
 // AzureClient This is an authorized client for Azure communication.
 type AzureClient struct {
+	config Config
 	compute.VirtualMachinesClient
-	compute.DisksClient
-	compute.VirtualMachineScaleSetsClient
 	network.InterfacesClient
+	network.LoadBalancersClient
 }
 
 // NewClient Initialized an authorized Azure client
-func NewClient(config Config) (client AzureClient) {
-	client = AzureClient{
-		compute.NewVirtualMachinesClientWithBaseURI(config.EnvironmentEndpoint, config.SubscriptionID),
-		compute.NewDisksClientWithBaseURI(config.EnvironmentEndpoint, config.SubscriptionID),
-		compute.NewVirtualMachineScaleSetsClientWithBaseURI(config.EnvironmentEndpoint, config.SubscriptionID),
-		network.NewInterfacesClientWithBaseURI(config.EnvironmentEndpoint, config.SubscriptionID),
+func NewClient(configMode string) (client AzureClient) {
+	var configload Config
+	if configMode == "metadata" {
+		configload = LoadConfig()
+	} else if configMode == "environment" {
+		configload = EnvLoadConfig()
+	} else {
+		log.Print("Invalid config Mode")
 	}
 
-	// Authorizing with Managed Service Identity
-	err := os.Setenv("AZURE_ENVIRONMENT", config.AzureEnvironment)
-	if err != nil {
-		log.Panicf("Unable to set Azure environment: %v", err)
-	}
-	authorizer, err := auth.NewAuthorizerFromEnvironment()
-	if err == nil {
-		client.VirtualMachinesClient.Authorizer = authorizer
-		client.DisksClient.Authorizer = authorizer
-		client.VirtualMachineScaleSetsClient.Authorizer = authorizer
-		client.InterfacesClient.Authorizer = authorizer
+	client = AzureClient{
+		configload,
+		GetVMClient(configload),
+		GetNicClient(configload),
+		GetLbClient(configload),
 	}
-
-	client.VirtualMachinesClient.RetryAttempts = 1
-	client.DisksClient.RetryAttempts = 1
-	client.VirtualMachineScaleSetsClient.RetryAttempts = 1
-	client.InterfacesClient.RetryAttempts = 1
-
 	return
 }
 
-// GetVM Returns a VirtualMachine object.
-func (c AzureClient) GetVM(vmname string) (vm compute.VirtualMachine) {
-	ctx, cancel := context.WithTimeout(context.Background(), 6000*time.Second)
-	defer cancel()
-
-	vm, err := c.VirtualMachinesClient.Get(ctx, Conf.ResourceGroup, vmname, compute.InstanceView)
+//GetVMClient return vmClient
+func GetVMClient(configload Config) compute.VirtualMachinesClient {
+	vmClient := compute.NewVirtualMachinesClient(configload.SubscriptionID)
+	a, err := auth.NewAuthorizerFromEnvironment()
 	if err != nil {
-		log.Panicf("failed to get VM: %v", err)
+		glog.Fatalf("failed to create authorizer from environment: %s\n", err)
```

Comment:
Just realized this, why are we switching from log to glog here?

Reply:
Not intentionally .

Reply:
Could we use log in that case to be consistent? :)

Reply:
There's a few other occasions where code was changed to glog I think.

Reply:
Sure I can use .